assets/services/illuminators/illuminator.py

```diff
@@ -0,0 +1,202 @@
+import os
+import threading
+import time
+from enum import Enum
+
+from pymavlink import mavutil
+
+
+class IlluminatorStatusFields(Enum):
+    UPTIME_MS = 0
+    ENABLE = 1
+    MODE_BITMASK = 2
+    ERROR_STATUS = 3
+    MODE = 4
+    BRIGHTNESS = 5
+    STROBE_PERIOD = 6
+    STROBE_DUTY_CYCLE = 7
+    TEMP_C = 8
+    MIN_STROBE_PERIOD = 9
+    MAX_STROBE_PERIOD = 10
+
+
+def stream_heartbeat(event, mavlink_connection_out):
+    while True:
+        event.wait()
+        mavlink_connection_out.mav.heartbeat_send(
+            mavutil.mavlink.MAV_TYPE_ILLUMINATOR, mavutil.mavlink.MAV_AUTOPILOT_INVALID, 0, 0, 0
+        )
+        time.sleep(1)
+
+
+def stream_illuminator_status(event, mavlink_connection_out):
+    while True:
+        event.wait()
+        mavlink_connection_out.mav.illuminator_status_send(*illuminator_status)
+        time.sleep(1)
+
+
+def publish_iluminator_status(mavlink_connection_out):
+    mavlink_connection_out.mav.illuminator_status_send(*illuminator_status)
+
+
+def publish_component_information_basic(mavlink_connection_out):
+    mavlink_connection_out.mav.component_information_basic_send(
+        0,  # time_boot_ms
+        0,  # capabilities
+        0,  # time_manufacture_s
+        "Illuminator Vendor".encode(),  # vendor_name
+        "Illuminator Model".encode(),  # model_name
+        "Illuminator Software Version".encode(),  # software_version
+        "Illuminator Hardware Version".encode(),  # hardware_version
+        "Illuminator Serial Number".encode(),  # serial_number
+    )
+
+
+def update_illuminator_status(field, value):
+    global illuminator_status
+    tmp = list(illuminator_status)
+    tmp[field.value] = value
+    illuminator_status = tuple(tmp)
+
+
+def handle_command_long(msg, mavlink_connection_out):
+    # MAV_CMD_REQUEST_MESSAGE
+    if msg.command == mavutil.mavlink.MAV_CMD_REQUEST_MESSAGE:
+        if msg.param1 == mavutil.mavlink.MAVLINK_MSG_ID_ILLUMINATOR_STATUS:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+            publish_iluminator_status(mavlink_connection_out)
+        elif msg.param1 == mavutil.mavlink.MAVLINK_MSG_ID_COMPONENT_INFORMATION_BASIC:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+            publish_component_information_basic(mavlink_connection_out)
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_UNSUPPORTED
```

Comment:
```suggestion
                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
```

Reply:
@vivian-zhou-skydio As Julian says, UNSUPPORTED is what you send if you don't understand a message at all and have no handling for it. [MAV_RESULT_DENIED](https://mavlink.io/en/messages/common.html#MAV_RESULT_DENIED) is for command supported but params are not.